frontend/catalyst/jit.py

```diff
@@ -129,12 +129,13 @@ def qjit(
         disable_assertions (bool): If set to ``True``, runtime assertions included in
             ``fn`` via :func:`~.debug_assert` will be disabled during compilation.
         seed (Optional[Int]):
-            The seed for random operations in a qjit call, such as circuit measurement results.
+            The seed for mid-circuit measurement results when the qjit-compiled function is executed
+            on simulator devices including ``lightning.qubit`` and ``lightning.kokkos``.
             The default value is None, which means no seeding is performed, and all processes
             are random. A seed is expected to be an unsigned 32-bit integer.
-            Note that `lightning.qubit` and `lightning.kokkos` currently only support seeding
-            measurements, and do not yet support seeding samples. As such, these devices with
-            shots will still return random results.
+            Note that seeding samples on simulator devices is not yet supported. As such,
+            shot-noise stochasticity in terminal measurement statistics such as ``sample`` or
+            ``expval`` will remain.
```

Comment:
I found this a bit confusing/unintuitive as a user when testing the seeding functionality :(

Reply:
As a result of this, `mcm_method="one-shot"` and `seed=something` _also_ results in randomness.

Reply:
The only thing that is seeded at the moment is `measure` (that is the same sequence of measurement results should appear everytime). I would be hesitant of advertising this to users yet.

Reply:
> The only thing that is seeded at the moment is `measure`

when `mcm_method="single-branch-statistics`

Reply:
What do you mean? I think the entire sequence of measurement results produced by the n shots of `dynamic-one-shot` would still be repeatable. What's important to note is that the seed is relative to the qjit program, not a qnode or tape execution, quantum device,  etc.

Reply:
Ah, I think my confusion is that if I switch `mcm_method` to `one-shot`, the results differ. Using the same example in the changelog but with one shot, I get this:

```python
Array([[0.7, 0.3],
       [0.7, 0.3],
       [0.7, 0.3],
       [0.7, 0.3]], dtype=float64)
```

I get this every run (note that as I have been running this with shots, I have 'removed' the shot noise in the results above).

But I think I am just confused, and my comment above can be ignored?

Reply:
Got it, yeah 👍 It doesn't guarantee identical results for the "same circuit" executed in different ways, only that the random state is fixed at the start of compiled program and will thus evolve in the same manner every time.

Reply:
👍